lib/omniauth/ebay-oauth/user_info.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module OmniAuth
+  module EbayOauth
+    class UserInfo # :nodoc:
+      def initialize(info)
+        @info = info['GetUserResponse']['User']
+      end
+
+      def uid
+        @info['EIASToken']
+      end
+
+      def info
+        {
+          username:   @info['UserID'],
+          first_name: name.split.first,
+          last_name:  name.split.last,
+          email:      @info['Email'],
+          phone:      @info['RegistrationAddress']['Phone'],
+          country:    @info['RegistrationAddress']['Country']
+        }
+      end
+
+      def extra
+        @info
```

Comment:
Same [OmniAuth guidelines](https://github.com/omniauth/omniauth/wiki/Auth-Hash-Schema#schema-10-and-later) prescribe, that all gathered raw info should be placed in `raw_info` key. Like this:

```json
{"extra":{"raw_info":{"AboutMePage":"false","EIASToken":"…"}}}
```